app/validators/mac_address_validator.rb

```diff
@@ -6,7 +6,10 @@ def validate_each(record, attribute, value)
   end
 
   def make_invalid(record, attribute)
-    record.errors.add(attribute, (options[:message] || _("is not a valid MAC address")))
+    # error message can already be present from the Net::Validations::normalize_mac method
+    if record.errors[attribute].empty?
```

Comment:
Do you happen to know if it's guaranteed to be non-nil? I'd be tempted to add the safe navigator.
```suggestion
    if record.errors[attribute]&.empty?
```

Reply:
I think it should always be non-nil, but we can't do anything wrong by adding the safe navigator so I am accepting the suggestion.

Reply:
Actually, I realize this inverts the logic should record.errors be nil. I will come up with something else.

Reply:
Added check for nil.